cadquery/occ_impl/assembly.py

```diff
@@ -290,3 +294,145 @@ def toJSON(
+    """
+    Converts the assembly to a fused compound and saves that within the document
+    to be exported in a way that preserves the face colors. Because of the use of
+    boolean operations in this method, performance may be slow in some cases.
+
+    :param assy: Assembly that is being converted to a fused compound for the document.
+    :param assy_name: Label that is applied to the top level object in the document.
+    """
+
+    # Prepare the document
+    app = XCAFApp_Application.GetApplication_s()
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+    app.InitDocument(doc)
+
+    # Shape and color tools
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+
+    # To fuse the parts of the assembly together
+    fuse_op = BRepAlgoAPI_Fuse()
+    args = TopTools_ListOfShape()
+    tools = TopTools_ListOfShape()
+
+    # If there is only one solid, there is no reason to fuse, and it will likely cause problems anyway
+    top_level_shape = None
+    if len(assy.children) == 1 and len(assy.children[0].shapes) == 1:
+        top_level_shape = assy.children[0].shapes[0].wrapped
+    else:
+        # Add base shape(s)
+        for shape in assy.children[0].shapes:
+            args.Append(shape.wrapped)
+
+        # Add all other shapes as tools
+        i = 0
+        for child in assy.children:
+            # Make sure we add the assembly parts other than the base
+            if i == 0:
+                i += 1
+                continue
+
+            for shape in child.shapes:
+                tools.Append(shape.wrapped)
+
+            i += 1
+
+        fuse_op.SetArguments(args)
+        fuse_op.SetTools(tools)
+        fuse_op.Build()
+
+        top_level_shape = fuse_op.Shape()
+
+    # If the top level shape is still none, something is wrong
+    if top_level_shape == None:
+        raise Exception("Error: The top level shape of assembly " + assy_name + " is not valid.")
+
+    # Add the fused shape as the top level object in the document
+    top_level_lbl = shape_tool.AddShape(top_level_shape, False)
+    TDataStd_Name.Set_s(top_level_lbl, TCollection_ExtendedString(assy_name))
+
+    # Walk the assembly->part->shape->face hierarchy and add subshapes for all the faces
+    for part in assy.children:
+        for shape in part.shapes:
+            for face in shape.Faces():
+                # Check for modified faces
+                modded_list = fuse_op.Modified(face.wrapped)
+
+                # If there are no modified faces associated with this face, treat it as-is
+                if modded_list.Size() == 0:
+                    # Add the face as a subshape
+                    cur_lbl = shape_tool.AddSubShape(top_level_lbl, face.wrapped)
+
+                    # If we got a null label, it should be because the face is covered by IsDeleted or Generated
+                    if cur_lbl.IsNull() == True:
```

Comment:
`== True` is not needed

Reply:
Avoided this after refactor using `IsSubShape()`.